### DISCUSSION TITLE
Thoughts about adding `map` for `tokio::sync::mpsc::error::TrySendError` (and/or others)

### DISCUSSION THREAD

**Discussion by winestone** [tokio-rs/tokio#4619] [Ideas]
Hi, would there be any objections or issues with adding `map` to `tokio::sync::mpsc::error::TrySendError` or other similar types? If not, I'm happy to make the relevant PRs.

Hopefully this isn't a duplicate, I've searched the issues and discussions here but haven't found anything relevant.

**Comment by Darksonn** [comment]
Why would you want that?

**Reply by winestone** [reply]
Thanks for the quick reply! I think sometimes you may want to add more info or remove information and it may be nicer than doing a `match`.

The particular use case I had was that I wanted to create a wrapper around `tokio::sync::mpsc::Sender<T>` but which instead sends `U`s, leading to me mapping over the error.

But of course there is a cost in that you have to maintain it into the future so I understand if the costs outweigh the rare use-cases.

**Reply by Darksonn** [reply]
I think this is too niche. You can always define your own method for doing it.
